### PR TITLE
3 -- T357971 and T362514: Refactor Example env vars, etc

### PR DESCRIPTION
**Pull request by lorenjohnson**
Merge https://github.com/wmde/wikibase-release-pipeline/pull/654 first, then change base to `main`

### CODE REVIEW

File: example/template.env [hunk 1]
```diff
@@ -1,55 +1,57 @@
...
-## Database Configuration
+# Database configuration
 DB_NAME=my_wiki
 DB_USER=sqluser
-DB_PASS=change-this-sqlpassword
+DB_PASS=change-this-password
+
+# The publically accessible hostnames which point to this host. You can instead just use the ports to access the services,
+# but without valid publically acceessible addresses QuickStatements will not be able to OAuth with Wikibase.
+WIKIBASE_PUBLIC_HOST=wikibase.local.gd
```
Comment (lorenjohnson) [inline example/template.env:26]: Rest the defaults to wikibase.mydomain.net, etc

File: example/template.env [hunk 2]
```diff
@@ -1,55 +1,57 @@
...
+WDQS_FRONTEND_PUBLIC_HOST=wdqs-frontend.local.gd
+QUICKSTATEMENTS_PUBLIC_HOST=quickstatements.local.gd
 
-## Wikibase Configuration
+# Ports published on host for front-end services. These probably only need to be changed if
```
Comment (lorenjohnson) [inline example/template.env:30]: Change to "The port numbers used for the front-end services. These can be reset to any unique set of port numbers that don't conflict with ports already being used on the host. Port 80 and 443 are used by the nginx-proxy reverse proxy service if that is not disabled. You can access the respective services directly through these ports (e.g. http://localhost:8880) on the host machine, or through the proxied public addresses above if configured."

**Comment by lorenjohnson** [comment]
Hey @RickiJay-WMDE, I found a couple more spots I in `example/docker-compose.yml`  for a good use of `depends_on` / `service_healthy`, and slightly better organised the WDQS related entries. Sorry if you're mid-review 😅

File: example/docker-compose.yml [hunk 3]
```diff
@@ -1,117 +1,108 @@
...
-      LETSENCRYPT_HOST: "${WIKIBASE_HOST}"
+      # Setting to 0 effectively disables jobs running after requests which is best when a jobrunner is used
+      MW_WG_JOB_RUN_RATE: 0
+      WIKIBASE_PINGBACK:
+      QS_PUBLIC_SCHEME_HOST_AND_PORT: http://${QUICKSTATEMENTS_PUBLIC_HOST}
+      VIRTUAL_HOST: ${WIKIBASE_PUBLIC_HOST}
+      LETSENCRYPT_HOST: ${WIKIBASE_PUBLIC_HOST}
     healthcheck:
-      test: [CMD, curl, --silent, --fail, "http://localhost/wiki/Main_Page"]
+      test: curl --silent --fail http://localhost/wiki/Main_Page
```
Comment (RickiJay-WMDE) [inline example/docker-compose.yml:44]: Can we refactor the other healthchecks like this? It looks a lot cleaner

File: example/template.env [hunk 4]
```diff
@@ -1,55 +1,68 @@
...
+#
+# Jobrunner Configuration
+MAX_JOBS=1
+#
+# Service Port Configuration
+# The ports specified below are the ports each front-end service will be published to on the host.
+# These only need to be changed if they are in conflict with existing services on the host.
+# The Reverse Proxy service (nginx-proxy) will also be on the HTTP and HTTPS ports (80 and 443).
+# NOTE: Currently default ports for MariaDB/MySQL (3306), Elasticsearch (9200), and WDQS (9999) are
+# set to standard defaults and not configurable using this template. If it is ncessary to change
```
Comment (RickiJay-WMDE) [inline example/template.env:53]: spelling - necessary

**Review by RickiJay-WMDE — CHANGES_REQUESTED** [review]

File: example/template.env (hunk 4 above)
Comment (RickiJay-WMDE) [inline example/template.env:53]: Why make them not configurable?

Comment (lorenjohnson) [reply example/template.env:53]: They were not before, this is a documentation update just noting that, and nlike the 3 front-end services which we have configurable ports for, these 2 have recognised "standard" defaults.  Also, we have a lot more considerations to make on the documentation bits here, but we have 3 tickets coming-up for that. The doc updates here are just a rough first pass and putting some stuff in-order on the way...

**Review by RickiJay-WMDE — APPROVED** [review]

**Pull request merged**